configure.ac

```diff
@@ -45,13 +45,15 @@ AC_ARG_ENABLE([debug],
 	      [AS_HELP_STRING([--enable-debug],
 			      [Enable debugging @<:@default=no@:>@])
 	      ],
-	      [CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} $CFLAGS"
-	       dbg=1],
-	      [enable_debug=no
-               dbg=0])
-
-AC_DEFINE_UNQUOTED([ENABLE_DEBUG],[$dbg],
-                   [defined to 1 if libfabric was configured with --enable-debug, 0 otherwise])
+	      [],
+	      [enable_debug=no])
+
+AS_IF([test x"$enable_debug" != x"no"],
+      [AC_DEFINE_UNQUOTED([ENABLE_DEBUG],[1],
+                   [defined to 1 if libfabric was configured with --enable-debug, 0 otherwise]),
+       CFLAGS="-g -O0 ${base_c_warn_flags} ${debug_c_warn_flags} $CFLAGS"],
+      [AC_DEFINE_UNQUOTED([ENABLE_DEBUG],[0],
+                   [defined to 1 if libfabric was configured with --enable-debug, 0 otherwise])])
```

Comment:
Thanks for doing this -- I saw #3531 this morning, but had to file #3534 first.  You beat me to fixing #3531.  👍 

Perhaps this might be a little less duplicated code:

```sh
AS_IF([test x"$enable_debug" != x"no"],
    [dbg=1
     CFLAGS="...."],
    [dbg=0])
AC_DEFINE_UNQUOTED([ENABLE_DEBUG], [$dbg],
    [defined to 1 if ...])
```

Reply:
Thanks for commenting this.
Your approach makes this more attractive 👍 
I've applied your comment and refreshed the PR.